core_tools/GUI/keysight_videomaps/liveplotting.py

```diff
@@ -102,6 +102,9 @@ def __init__(self, pulse_lib, digitizer, scan_type = 'Virtual', cust_defaults =
             from core_tools.GUI.keysight_videomaps.data_getter import scan_generator_Tektronix
             self.construct_1D_scan_fast = scan_generator_Tektronix.construct_1D_scan_fast
             self.construct_2D_scan_fast = scan_generator_Tektronix.construct_2D_scan_fast
+        elif isinstance( scan_type , dict):
```

Comment:
Using a dictionary to set the scan functions is a pretty dirty way to do this. This hook is not documented.
A proper way to solve it is with a base class that defines the interfaces for the fast scan and a second class that contains the implementations for 1D and 2D.

Reply:
@sldesnoo-Delft Fair point. I replaced the dict with an abstract base class. Docstring is copied from the Keysight backend.